### PR TITLE
Issue 88

### PR DESCRIPTION
**Pull request by ashraffouda**

### CODE REVIEW

File: lib/JumpScale/clients/cockpit/CockpitBot.py [hunk 1]
```diff
@@ -148,3 +156,46 @@ def run(self):
...
+            self._check_services(chat_id, username, get_status=True)
+        except Exception as e:
+            self.logger.error(e)
+            self.bot.sendMessage(chat_id=chat_id, text="Error happened %s" % e)
+
+    def _check_services(self, chat_id, username, get_status=False):
+        states = []
+        for file in glob.glob("cockpit/%s/services/**/*.yaml" % username, recursive=True):
+            with open(file, 'r') as f:
+                y = yaml.load(f)
```
Comment (zaibon) [inline lib/JumpScale/clients/cockpit/CockpitBot.py]: There is easier way to inspect the status of a service then parsing manually the state file.
A service object has a state object attached to it.
e.g:
```
repo =  j.atyourservice.repos['cockpit']
service = repo.findService('...')
service.state.methods
```
`service.state.methods` returns a dict that contains the status for all methods of the service.
````
{'init': 'OK', 'install' : 'OK'}
```

File: lib/JumpScale/clients/cockpit/Asker.py [hunk 2]
```diff
@@ -269,6 +269,30 @@ def validate(input):
         portal_password = self.askChoice("Please enter a portal password")
         return portal_password
 
+    def ask_organization(self):
+        def validate(input):
+            return True
+        organization = self.askString("Please enter organization name at itsyou.online")
+        return organization
+
+    def ask_client_id(self):
```
Comment (zaibon) [inline lib/JumpScale/clients/cockpit/Asker.py]: Why do we ask client id and secret to the user directly ?

File: lib/JumpScale/clients/cockpit/Asker.py [hunk 3]
```diff
@@ -269,6 +269,30 @@ def validate(input):
...
+        def validate(input):
+            return True
+        client_id = self.askString("Please enter the client ID")
+        return client_id
+
+    def ask_client_secret(self):
```
Comment (zaibon) [inline lib/JumpScale/clients/cockpit/Asker.py]: same

File: lib/JumpScale/clients/cockpit/Asker.py [hunk 4]
```diff
@@ -269,6 +269,30 @@ def validate(input):
...
+        def validate(input):
+            return True
+        client_secret = self.askString("Please enter the client secret")
+        return client_secret
+
+    def ask_ityou_username(self):
```
Comment (zaibon) [inline lib/JumpScale/clients/cockpit/Asker.py]: same

File: lib/JumpScale/clients/cockpit/CockpitDeployer.py [hunk 5]
```diff
@@ -451,3 +471,19 @@ def deploy(self):
         container_cuisine.core.file_write('/root/.ssh/authorized_keys', key_pub)
 
         return cockpit_repo_path
+
+    def oauth(self, chat_id):
```
Comment (zaibon) [inline lib/JumpScale/clients/cockpit/CockpitDeployer.py]: I don't get this. Why do we do a request on the flask server ouself.
flask server is only there as a call back endpoint for it'syou.online. we don't need to query it ourself.

Comment (ashraffouda) [inline lib/JumpScale/clients/cockpit/CockpitDeployer.py]: this for the server to construct the url cuz the server will keep track for this request

File: lib/JumpScale/clients/cockpit/Asker.py (hunk 2 above)
Comment (ashraffouda) [inline lib/JumpScale/clients/cockpit/Asker.py]: according to this script https://gist.github.com/robvanmieghem/d8606a18aec0b495a147d2ce8f0bc533
we need the client secret to get the user token

Comment (zaibon) [inline lib/JumpScale/clients/cockpit/Asker.py]: client secret and client_id are private for the bot. these are not information that you get from the user.

**Comment by rkhamis** [comment]
@ashraffouda @zaibon what's the status of this PR?

**Pull request closed without merging**